### ISSUE TITLE
Prevent highlighting of escape characters in character arrays

### ISSUE THREAD

**Issue by goldrik** [mathworks/MATLAB-extension-for-vscode#79]
**Describe the bug**
When initializing a raw string variable, escape characters (backslash + char) get highlighted by default. This occurs for character arrays (using single quotes), but not for strings (using double quotes).

For example, when defining a Windows path:
`temp_path = '\\somedrive\path\to\right\folder\';`
In this case, the escape characters `\\`, `\r`, `\t`, and `\f` characters would be highlighted. 

If you do the same but with double quotations, the error does not occur. This is the expected behavior.

**Comment by dklilley** [comment]
Hi @goldrik , thanks for reporting this!

For clarity, can you please attach an image showing this behavior?

**Comment by goldrik** [comment]
Yup, here's the same string, initialized as a character array, then as a string. 

<img width="338" alt="Screenshot 2024-01-05 at 1 52 34 PM 2" src="https://github.com/mathworks/MATLAB-extension-for-vscode/assets/8674612/08cf598d-f2ee-4782-ba87-1b658428978b">

**Comment by dklilley** [comment]
I don't think it is a bug that the escape characters are being highlighted in character vectors - they are explicitly being tokenized as `constant.character.escape.matlab`. It is definitely incorrect, however, that they are being treated differently between char vectors and string arrays. 

I am moving this issue over to the MATLAB-Language-grammar repository, because this is an issue with the grammar.
